pq-crypto/s2n_kyber_512_evp.c

```diff
@@ -24,9 +24,9 @@
 
 #if defined(S2N_LIBCRYPTO_SUPPORTS_KYBER512)
 int s2n_kyber_512_evp_generate_keypair(uint8_t *public_key, uint8_t *private_key) {
-    EVP_PKEY_CTX *kyber_pkey_ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_KYBER512, NULL);
+    EVP_PKEY_CTX *kyber_pkey_ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_KEM, NULL);
```

Comment:
Do you mind fixing my memory leak while you're here? My C was a bit rusty when I originally authored this and totally forgot about memory management. The `EVP_PKEY_CTX` needs to be freed at the end of each routine. Similarly, `EVP_PKEY` needs to be dealt with on error cases.

Reply:
thank you both for the feedback.

it looks like `POSIX_ENSURE` and `POSIX_GUARD_PTR` are [deprecated](https://github.com/aws/s2n-tls/blob/main/utils/s2n_safety_macros.h#L210). shall I take this opportunity to migrate this file to use `s2n_result.h`?

Reply:
> Similarly, EVP_PKEY needs to be dealt with on error cases.

to my understanding, this is [handled](https://github.com/aws/s2n-tls/blob/main/utils/s2n_safety_macros.h#L200-L214) by the `POSIX_ENSURE` and `POSIX_GUARD_PTR` macro calls.

Reply:
Yeah, I love the cleanup attribute / DEFER_CLEANUP. But if I remember right it's not available in msvc / Windows. So s2n-tls will probably never support Windows :)